platform/src/main/java/org/hillview/table/SortedStringsConverterDescription.java

```diff
@@ -61,17 +61,22 @@ public IStringConverter getConverter() {
 
     public class Converter implements IStringConverter {
         private final Object2DoubleOpenHashMap<String> memoizedResults;
+        // The smallest value given by computeIndex is -1.
+        private final double keyNotFound;
+        private static final double EPSILON = 0.01;
 
         public Converter() {
             this.memoizedResults = new Object2DoubleOpenHashMap<String>();
+            this.keyNotFound = (boundaries.length) + 2;
         }
 
         @Override
         public double asDouble(@Nullable String string) {
-            if (memoizedResults.containsKey(string)) {
-                return memoizedResults.getDouble(string);
+            double index = memoizedResults.getOrDefault(string, keyNotFound);
+            if (index < keyNotFound - EPSILON) {
```

Comment:
I don't get the epsilon.

Reply:
Was worried about floating point precision problems.

Reply:
there is no arithmetic on these fields, so that should not be a problem
we often compare doubles for equality, e.g. in heavy hitters or hyperLogLog